jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -234,35 +232,42 @@ private void authenticatePush() {
   }
 
   private void saveDocker() {
-    ProgressEventDispatcher.Factory childProgressDispatcherFactory =
-        Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
-
     Optional<DockerClient> dockerClient =
         buildConfiguration.getBaseImageConfiguration().getDockerClient();
     Preconditions.checkArgument(dockerClient.isPresent());
-
-    results.tarPath =
+    ProgressEventDispatcher.Factory childProgressDispatcherFactory =
+        Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
+    Future<LocalImage> localImageFuture =
+        executorService.submit(
+            () ->
+                new LocalBaseImageSteps(
+                        executorService, buildConfiguration, childProgressDispatcherFactory)
+                    .dockerDaemonImageStep(dockerClient.get()));
+    results.baseImageAndAuth =
+        executorService.submit(
+            () -> new ImageAndAuthorization(localImageFuture.get().baseImage, null));
+    results.baseImageLayers =
         executorService.submit(
-            new SaveDockerStep(
-                buildConfiguration,
-                dockerClient.get(),
-                childProgressDispatcherFactory,
-                tempDirectoryProvider));
+            () ->
+                localImageFuture
+                    .get()
+                    .layers
+                    .stream()
+                    .map(Futures::immediateFuture)
+                    .collect(Collectors.toList()));
   }
 
   private void extractTar() {
+    Optional<Path> tarPath = buildConfiguration.getBaseImageConfiguration().getTarPath();
+    Preconditions.checkArgument(tarPath.isPresent());
     ProgressEventDispatcher.Factory childProgressDispatcherFactory =
         Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
     Future<LocalImage> localImageFuture =
         executorService.submit(
             () ->
-                new ExtractTarStep(
-                        executorService,
-                        buildConfiguration,
-                        results.tarPath.get(),
-                        childProgressDispatcherFactory,
-                        tempDirectoryProvider)
-                    .call());
+                new LocalBaseImageSteps(
+                        executorService, buildConfiguration, childProgressDispatcherFactory)
+                    .tarImageStep(tarPath.get()));
```

Comment:
This name is also odd as the other `Step`s are classes that are `Callable`, whereas this produces a result.